packages/gatsby/src/schema/__tests__/queries.js

```diff
@@ -116,10 +116,10 @@ describe(`Query schema`, () => {
                     query: {
                       filter: {
                         frontmatter: {
-                          authors: { email: { eq: source.email } },
-                          // authors: {
-                          //   elemMatch: { email: { eq: source.email } },
-                          // },
+                          // authors: { email: { eq: source.email } },
+                          authors: {
+                            elemMatch: { email: { eq: source.email } },
+                          },
```

Comment:
Note: I've verified with @freiksenet that this test was broken and should not have relied on `eq` working on arrays. Heck, the equivalent `elemMatch` code was still here anyways, so reverting to that.